src/modules/thirdParty/sagas.ts

```diff
@@ -225,30 +236,58 @@ export function* thirdPartySaga(builder: BuilderAPI, catalyst: CatalystClient) {
     }
   }
 
+  function* handleUpdateApprovalFlowProgress(action: { progress: number }) {
+    yield put(updateApprovalFlowProgress(action.progress))
+  }
+
   function* handleDeployBatchedThirdPartyItemsRequest(action: DeployBatchedThirdPartyItemsRequestAction) {
     const { items, collection, tree, hashes } = action.payload
     const REQUESTS_BATCH_SIZE = 5
 
     let queue = new PQueue({ concurrency: REQUESTS_BATCH_SIZE })
-    try {
-      const identity: AuthIdentity | undefined = yield call(getIdentity)
+    const errors: ThirdPartyError[] = []
+    const identity: AuthIdentity | undefined = yield call(getIdentity)
 
-      if (!identity) {
-        throw new Error('Invalid Identity')
-      }
+    if (!identity) {
+      yield put(deployBatchedThirdPartyItemsFailure(errors, 'Invalid Identity'))
+      return
+    }
 
-      const promisesOfItemsBeingDeployed: (() => Promise<ItemCuration>)[] = items.map((item: Item) => async () => {
-        const entity: DeploymentPreparationData = await buildTPItemEntity(catalyst, builder, collection, item, tree, hashes[item.id])
-        await catalyst.deployEntity({ ...entity, authChain: Authenticator.signPayload(identity, entity.entityId) })
-        return builder.updateItemCurationStatus(item.id, CurationStatus.APPROVED)
+    yield put(
+      openModal('ApprovalFlowModal', {
+        view: ApprovalFlowModalView.DEPLOYING_TP
       })
+    )
+    const promisesOfItemsBeingDeployed: (() => Promise<ItemCuration | void>)[] = items.map((item: Item) => async () => {
+      let entity: DeploymentPreparationData
+      try {
+        entity = await buildTPItemEntity(catalyst, builder, collection, item, tree, hashes[item.id])
+        try {
+          await catalyst.deployEntity({ ...entity, authChain: Authenticator.signPayload(identity!, entity.entityId) })
+          approvalFlowProgressChannel.put({
+            progress: Math.round(((items.length - (queue.size + queue.pending)) / items.length) * 100)
+          })
+          let updatedCuration: ItemCuration
+          try {
+            updatedCuration = await builder.updateItemCurationStatus(item.id, CurationStatus.APPROVED)
+            return updatedCuration
+          } catch (error) {
+            errors.push(new ThirdPartyCurationUpdateError(item))
+          }
+        } catch (error) {
+          errors.push(new ThirdPartyDeploymentError(item))
+        }
+      } catch (error) {
+        errors.push(new ThirdPartyBuildEntityError(item))
+      }
+      return Promise.resolve()
```

Comment:
As an optional comment, the return here might be redundant.
```suggestion
      return Promise.resolve()
```